pkg/endpoint/endpoint.go

```diff
@@ -2295,3 +2300,27 @@ func (e *Endpoint) PinDatapathMap() error {
 
 	return err
 }
+
+// SyncEndpointHeaderFile it bumps the current DNS History information for the
+// endpoint in the lxc_config.h file.
+func (e *Endpoint) SyncEndpointHeaderFile() error {
+	e.UnconditionalLock()
+	defer e.Unlock()
```

Comment:
I think you should hold the `BuildMutex` here instead of the normal mutex. This is because while the endpoint's BPF program is being built / compiled, the endpoint's Lock is *not* held. I'm not sure what effects overwriting the file during compilation would have, but I think we should synchronize this to not occur while the endpoint is building (i.e., while the `BuildMutex` is being held). However, I'm just not sure what the side effects of this are :/ 

cc: @joestringer

Reply:
Right, we wouldn't want to get into a situation where we end up rewriting the `lxc_config.h` in the middle of a compile or something. 

We can add to the comment above the `e.dnsHistoryTrigger` field in the endpoint to describe that it is protected by the `BuildMutex` (as this is different from most fields in the endpoint which are protected by the endpoint mutex).